examples/kotlin/control/MainActivity.kt

```diff
@@ -0,0 +1,80 @@
+package io.envoyproxy.envoymobile.hellocontrolkotlin
+
+import android.app.Activity
+import android.content.Context
+import android.os.Bundle
+import android.os.Handler
+import android.os.HandlerThread
+import android.support.v7.widget.DividerItemDecoration
+import android.support.v7.widget.LinearLayoutManager
+import android.support.v7.widget.RecyclerView
+import android.util.Log
+
+import java.io.IOException
+import java.io.InputStream
+import java.net.HttpURLConnection
+import java.net.URL
+import java.util.concurrent.TimeUnit
+
+private const val REQUEST_HANDLER_THREAD_NAME = "hello_envoy_kt"
+private const val ENDPOINT = "http://s3.amazonaws.com/api.lyft.com/static/demo/hello_world.txt"
```

Comment:
I was using `jsonplaceholder.typicode.com/comments` to simulate a larger response.

```
# Envoy config used by the example applications.
# More information about Envoy's config can be found at:
# https://www.envoyproxy.io/docs/envoy/latest/configuration/configuration
static_resources:
  listeners:
    - address:
        socket_address: {address: 0.0.0.0, port_value: 9001, protocol: TCP}
      filter_chains:
        - filters:
            - name: envoy.http_connection_manager
              config:
                stat_prefix: client
                route_config:
                  virtual_hosts:
                    - name: all
                      domains: ["*"]
                      routes:
                        - match: {prefix: "/"}
                          route: {cluster: hello_world_api}
                http_filters:
                  - name: envoy.router
  clusters:
    - name: hello_world_api
      connect_timeout: 30s
      dns_lookup_family: V4_ONLY
      lb_policy: ROUND_ROBIN
      load_assignment:
        cluster_name: hello_world_api
        endpoints:
          - lb_endpoints:
              - endpoint:
                  address:
                    socket_address: {address: jsonplaceholder.typicode.com, port_value: 443}
      tls_context:
        sni: jsonplaceholder.typicode.com
      type: LOGICAL_DNS

```